gallery/src/components/NxNumberInput/NxNumberInputComplexExample.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useState } from 'react';
+
+import { NxNumberInput, NxNumberInputStateProps, nxTextInputStateHelpers } from '@sonatype/react-shared-components';
+
+const { initialState, userInput } = nxTextInputStateHelpers;
+
+export default function NxNumberInputSimpleExample() {
+  // Note that NxNumberInputStateProps is the TypeScript type for the state helper function return types.
+  // The explicity type parameter isn't necessary here, it's only present to demonstrate that type.
+  const [state, setState] = useState<NxNumberInputStateProps>(initialState(''));
```

Comment:
I think this comment and the type parameter are better off omitted

Reply:
> I think this comment and the type parameter are better off omitted. Same in the other examples

https://github.com/sonatype/sonatype-react-shared-components/pull/486/commits/49f678587a1c0f3ff46f40563b3dd302c8f49dd6